### DISCUSSION TITLE
vercel.json functions configuration not applied to created functions

### DISCUSSION THREAD

**Discussion by lalish99** [vercel/community#6398] [Help] [answered]
### Summary

My account is currently subscribed to the Pro plan. So, according to [the docs](https://vercel.com/docs/functions/runtimes#max-duration) our functions should be able to have a max duration of up-to 300 seconds. But, I'm unable to configure the max duration or the memory of the functions of my deployment.

Here's my `vercel.json` file:
```JSON
{
    "functions": {
        "pages/portals/**/*": {
            "maxDuration": 20,
            "memory": 2048
        },
        "pages/topics/**/*": {
            "maxDuration": 20,
            "memory": 2048
        },
        "pages/api/revalidate.ts": {
            "maxDuration": 30
        }
    }
}
```

The folder structure looks like this:
<img width="290" alt="Screenshot 2024-04-04 at 11 15 47 a m" src="https://github.com/vercel/community/assets/38923878/d86b7605-c4bd-441f-a2fc-8ce6f7039ae7">

The deployment works as expected and I don't see any relevant errors on the build logs. But, after the deployment finishes, the properties of the functions are still the default values:

![Screenshot 2024-04-04 at 11 16 50 a m](https://github.com/vercel/community/assets/38923878/c1e4bb53-514a-4f20-8cab-10df546feb06)

Am I misconfiguring something?

### Steps to Reproduce

I'm not sure if this is replicable by others, but I have my `vercel.json` like this:

```JSON
{
    "functions": {
        "pages/portals/**/*": {
            "maxDuration": 20,
            "memory": 2048
        },
        "pages/topics/**/*": {
            "maxDuration": 20,
            "memory": 2048
        },
        "pages/api/revalidate.ts": {
            "maxDuration": 30
        }
    }
}
```

**Comment by amyegan** [comment]
Hey @lalish99. You can find some more info about configuring `maxDuration` at [vercel.com/docs/functions/configuring-functions/duration#](https://vercel.com/docs/functions/configuring-functions/duration). When viewing the code examples, you can choose other frameworks and switch between typescript or javascript. Make sure you have the right options selected so you see the recommended method for your framework.

<img width="825" alt="Screenshot" src="https://github.com/vercel/community/assets/8247105/e10d4fc4-723d-4dec-aa3a-731ec439f7ad">


I need to know a bit more about your project to be able to give a specific solution. If you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), that would let me see what framework and architecture is used.

**Comment by lalish99** [comment]
@amyegan  Our project is currently running on NextJS `^12.3.4`, that's why I was using the [Next.js versions older than 13.5](https://vercel.com/docs/functions/configuring-functions/duration#other-frameworks-and-runtimes-next.js-versions-older-than-13.5-go-python-or-ruby) option described in the same docs you shared. 

The problem is that the configuration isn't applied for any of the created functions. So, I was just wondering if `vercel.json` could get overwritten somehow? Or if there's why it could be ignored?

I've already read all of [this discussion](https://github.com/orgs/vercel/discussions/4391) trying to troubleshoot. But honestly, I'm stuck. There are no clear errors on the logs, and the deployment works as expected. But as you can see, the functions still have the default configuration. The function shown in the following screenshot should have a max duration of 20 seconds, and a memory size of 2048 MB based on the `vercel.json` I previously shared.

```JSON
{
    "functions": {
        "pages/portals/**/*": {
            "maxDuration": 20,
            "memory": 2048
        },
        "pages/topics/**/*": {
            "maxDuration": 20,
            "memory": 2048
        },
        "pages/api/revalidate.ts": {
            "maxDuration": 30
        }
    }
}
```

![Screenshot 2024-04-04 at 11 50 35 a m](https://github.com/vercel/community/assets/38923878/31c4b387-6f08-40da-89ca-6dc9fde95280)

**Reply by amyegan** [reply]
Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that I can use to help you debug this?

**Reply by lalish99** [reply]
No, I'll check with my team if we can create one. In the meantime, could you let me know if there is anything else that needs to be configured for the `vercel.json` functions configuration to be applied?

**Reply by lalish99** [reply]
For example, does [the settings function configuration](https://vercel.com/docs/functions/configuring-functions/memory) limit the capacity of the functions? even with the Pro plan?

![Screenshot 2024-04-04 at 12 21 40 p m](https://github.com/vercel/community/assets/38923878/3a0f60b1-8c3b-45a3-b960-d0c780b3a2c9)

**Reply by amyegan** [reply]
As far as I know the configuration should already be applied as it is. But I don't know what else is configured for the project that could be interfering with that. Nothing I can think of at the moment should cause `vercel.json` to be ignored, but we can see from your results that something is going wrong.

**Reply by amyegan** [reply]
> Any [custom memory size](https://vercel.com/docs/functions/configuring-functions/memory#setting-custom-function-memory-size) set in the vercel.json file will take precedence over the default memory size set in the dashboard.

https://vercel.com/docs/functions/configuring-functions/memory#setting-your-default-function-memory-/-cpu-size

**Reply by lalish99** [reply]
Then I'm as confused as you are. I'll check with my team if we can do a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) and come back to you.

**Reply by lalish99** [reply]
@amyegan I have the [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) ready on a private repo. My team is asking if we could add you to the repo instead of making it public.

**Reply by swarnava** [reply]
Hi, we don't accept private repository unfortunately due to security & privacy reason.

**Reply by amyegan** [reply]
We ask for a **public** repo for a variety of reasons. The main one is that I may need to involve other engineers in problem investigation. Another reason is that another community member might be able to look into your issue and provide a solution sooner than I can. 

The MRE should use as little code as possible to show the problem and should not contain any sensitive info. This strategy lets you share a public example of the problem without exposing your website or app code. It also helps narrow down the possible causes of the issue.

**Reply by lalish99** [reply]
No problem, I've asked my team to make the repo public.

**Reply by lalish99** [reply]
Here's the repo

https://github.com/greenriver/delaware-fe-mre-vercel

The deployed function still has the default time-out value.
<img width="1418" alt="Screenshot 2024-04-05 at 12 56 20 p m" src="https://github.com/vercel/community/assets/38923878/2d8332d0-bb95-4044-99b9-a338926e2b05">

**Reply by lalish99** [reply]
@amyegan where you able to replicate the problem?

**Reply by amyegan** [reply, marked as answer]
I was able to see the problem and I have a solution. Your project uses a `/src` directory, so you'll need to include that In the `vercel.json` config. For example, if you change `"pages/api/revalidate.ts"` to `"src/pages/api/revalidate.ts"` you'll see the expected maximum execution duration reflected on a new deployment.
